### DISCUSSION TITLE
Symfony: Fatal Error in Dependency Injection with Custom Bundle

### DISCUSSION THREAD

**Discussion by sn00ns** [symfony/symfony#58900] [Q&A]
### **Title:** Symfony: Fatal Error in Dependency Injection with Custom Bundle

Hi Community,

I'm currently diving deeper into Symfony and working on my first custom bundle. While setting up the bundle was straightforward, I'm stuck with a **dependency injection issue** that I can't resolve on my own. Despite going through tutorials, using ChatGPT, and doing extensive research, I couldn't find a solution. I’m hoping for help from a Symfony expert.

### **Problem:** ###
When running `composer install` or accessing a route in the browser, I encounter the following error:
```plaintext  
Fatal error: Cannot declare class MoSys\Rithmo\DependencyInjection\RithmoExtension,  
because the name is already in use in /app/src/MoSys/Rithmo/DependencyInjection/RithmoExtension.php on line 8  
```
Symfony reports that the class RithmoExtension is being declared twice, but I can’t figure out why. The class is only defined once in my code:

**Namespace:** MoSys\Rithmo\DependencyInjection#  
**Class Name:** RithmoExtension

## Project Setup: ##

The complete Symfony project is located in the main branch of the repository, so you can see all the dependencies.
The bundle is located in ../src/MoSys/Rithmo/....

You can find the project on GitHub here::
GitHub Repository: https://github.com/sn00ns/rithmo

## Error Output (excerpt): ##
    !!  Fatal error: Cannot declare class MoSys\Rithmo\DependencyInjection\RithmoExtension, 
        because the name is already in use

### What I’ve Tried So Far: ###
1. Verified the file and namespace – they seem correct.
2. Regenerated Composer autoload files (composer dump-autoload).
3. Cleared the Symfony cache multiple times (bin/console cache:clear).
4. Went through various tutorials and documentation, but couldn’t pinpoint the issue.

### Question: ###
What could cause this error? Are there any common pitfalls or specific issues I might have overlooked?
I’d really appreciate any guidance. Thank you in advance!

**Comment by andrew-demb** [comment]
Check this update: https://github.com/sn00ns/rithmo/pull/1

1. You have two namespace roots configured for one file (`RithmoExtension.php`)
2. Symfony configured to auto-register all files under `src/`

https://github.com/sn00ns/rithmo/blob/609864097b351d70900ccfe8302f8c4551528e65/app/composer.json#L26-L31

https://github.com/sn00ns/rithmo/blob/609864097b351d70900ccfe8302f8c4551528e65/app/config/services.yaml#L16-L17

**Comment by sn00ns** [comment]
@andrew-demb 

**Dear Andrii,**

Thank you so much! I hope you are in a safe place.

I’ve been learning a lot recently, especially about Symfony. It wasn’t clear to me that everything in the src directory is automatically loaded, and I realized that I caused the duplicate loading issue myself.

This detail didn’t stand out to me while studying the documentation, which made your input even more valuable.

Would you be interested in an exchange regardings Symfony?

**Best regards,
Dirk**

**Reply by andrew-demb** [reply]
I'm glad to help.

> Would you be interested in an exchange regardings Symfony?

Not for now, sorry.
